lib/bloc/podcast_clip/podcast_clip_bloc.dart

```diff
@@ -0,0 +1,238 @@
+  }
+
+  Future<String> _getAudioClipPath(
+      {PodcastClipDetailsModel clipDetails}) async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+
+    String episodeUrl = clipDetails.episodeDetails.contentUrl;
+    String audioClipPath = await initClipDirectory(isVideoClip: false);
+    audioClipPath = audioClipPath + '/clip.mp3';
+
+    //Getting the raw audio clip
+    String clippedAudioCommand =
+        '-i "$episodeUrl" -ss ${clipDetails.startTimeStamp.inSeconds} -to ${clipDetails.endTimeStamp.inSeconds}  -acodec copy $audioClipPath';
+
+    FFmpegSession ffpegSessionDetails =
+        await FFmpegKit.execute(clippedAudioCommand);
+
+    final returnCode = await ffpegSessionDetails.getReturnCode();
+    if (ReturnCode.isSuccess(returnCode)) {
+      return audioClipPath;
+    } else {
+      return null;
+    }
+  }
+
+  getVideoClipPath({String audioClipPath, String episodeImagepath}) async {
+    String videoClipPath = await initClipDirectory(isVideoClip: true);
+    videoClipPath = videoClipPath + '/VideoClip.mp4';
+
+    String clippedVideoCommand =
+        '-i  $audioClipPath -i $episodeImagepath -filter_complex "[0:a]showwaves=colors=0xffffff@0.9:mode=p2p,format=yuva420p[v];[1:v]scale=2120:2560[bg];[bg][v]overlay=(main_w-overlay_w)/2:H*0.8[outv]" -map "[outv]" -map 0:a -c:v libx264 -c:a copy $videoClipPath';
+
+    FFmpegSession ffpegSessionDetails =
+        await FFmpegKit.execute(clippedVideoCommand);
+
+    String logs = await ffpegSessionDetails.getLogsAsString();
+    log('cliplogs $logs');
+    final returnCode = await ffpegSessionDetails.getReturnCode();
+    if (ReturnCode.isSuccess(returnCode)) {
+      return videoClipPath;
+    } else {
+      return null;
+    }
+  }
+
+  Future<bool> isEpisodeClipable() async {
+    var clipDetails = await getCurrentPodcastClipDetails();
+
+    if ((clipDetails.endTimeStamp + Duration(seconds: _initialSeconds)) >
+        clipDetails.episodeLength) {
+      return false;
+    }
+    return true;
+  }
+
+  Future<String> clipEpisode({String podcastImageWidgetPath}) async {
+    _clipStateBehaviourSubject.add(PodcastClipState.FETCHING_AUDIO_CLIP);
+    var clipDetails = await getCurrentPodcastClipDetails();
+    if (clipDetails == null) {
+      return null;
+    }
+    String audioClipPath = await _getAudioClipPath(clipDetails: clipDetails);
+
+    if (audioClipPath == null) {
+      //show error method
+      return null;
+    }
+
+    String videoClipPath = await getVideoClipPath(
+        audioClipPath: audioClipPath, episodeImagepath: podcastImageWidgetPath);
+
+    _clipStateBehaviourSubject.add(PodcastClipState.IDLE);
+    if (videoClipPath == null) {
+      return null;
+    }
+
+    return videoClipPath;
+  }
+
+  setPodcastState(PodcastClipState clipState) {
+    _clipStateBehaviourSubject.add(clipState);
+  }
+
+  Future<String> initClipDirectory({bool isVideoClip}) async {
+    Directory appDocDirectory = await getApplicationDocumentsDirectory();
```

Comment:
Maybe better to use `getTemporaryDirectory` ?

Reply:
Done